File: src/contests/cg/puzzle/fsharp/the-river-i-.fs
// 🎮 CodinGame Puzzle - the-river-i-
// https://www.codingame.com/training/easy/the-river-i-

let digitSum (x0: int64) =
    let mutable x = x0
    let mutable s = 0L
    while x > 0L do
        s <- s + x % 10L
        x <- x / 10L
    s

let mutable a = int64 (System.Console.ReadLine().Trim())
let mutable b = int64 (System.Console.ReadLine().Trim())
while a <> b do
    if a < b then a <- a + digitSum a else b <- b + digitSum b
printfn "%d" a

File: src/contests/cg/puzzle/kotlin/the-river-i-.kt
// 🎮 CodinGame Puzzle - the-river-i-
// https://www.codingame.com/training/easy/the-river-i-

fun digitSum(x0: Long): Long {
    var x = x0
    var s = 0L
    while (x > 0) {
        s += x % 10
        x /= 10
    }
    return s
}

fun main() {
    var a = readLine()!!.trim().toLong()
    var b = readLine()!!.trim().toLong()
    while (a != b) {
        if (a < b) a += digitSum(a) else b += digitSum(b)
    }
    println(a)
}
